nix/default.nix

```diff
@@ -46,16 +46,26 @@ let
   self = with pkgs; stdenv.mkDerivation rec {
     name = "crun";
     src = ./..;
+    vendorSha256 = null;
     doCheck = false;
     enableParallelBuilding = true;
-    nativeBuildInputs = [ autoreconfHook git go-md2man pkg-config python3 ];
-    buildInputs = [ criu glibc glibc.static libcap libseccomp protobufc systemd yajl ];
+    outputs = [ "out" ];
+    nativeBuildInputs = [ autoreconfHook bash git go-md2man pkg-config python3 which ];
+    buildInputs = [ glibc glibc.static criu libcap libseccomp protobufc yajl systemd ];
```

Comment:
do we need to drop systemd from here when it is optional?

Reply:
Yes it should be excluded, now updated as:
```
    buildInputs = [ glibc glibc.static criu libcap libseccomp protobufc yajl ]
      ++ lib.optional (!disableSystemd) [ systemd ];
    configureFlags = [ "--enable-static" ]
      ++ lib.optional (disableSystemd) [ "--disable-systemd"];
```

Reply:
@giuseppe I try to make the systemd optional, but crun `./configure` will ALWAYS check if that header exists, even though we say `--disable-ssytemd`.

Therefore already precompile with systemd library could solve the issue :-(